CMakeLists.txt

```diff
@@ -3,18 +3,14 @@ project(hwinfo VERSION 1.0.0 LANGUAGES CXX)
 
 if (NOT DEFINED NO_OCL)
     set(CMAKE_CXX_STANDARD 20)
-    set(CMAKE_CXX_STANDARD_REQUIRED ON)
 else ()
-    set(CMAKE_CXX_STANDARD 11)
-    set(CMAKE_CXX_STANDARD_REQUIRED ON)
+    set(CMAKE_CXX_STANDARD 17)
```

Comment:
In PR #32, we have actively decided to reduce the standard requirement from 17 to 11 even though, we need to implement some c++17 features.

If OpenCL is included, We currently need c++20 since I my OpenCL wrapper library which uses c++20. Anyways, I keep it on my list, to reduce this back to c++11 as well...

Reply:
The reason for changing to C++17 is that I changed the if/else construct in WMIMapper.h to use `if constexpr` (a C++17 feature) in order to resolve compiler warnings. In principle, this could also be done using `std::enable_if`, but that's very cumbersome.

It's of course your choice, but I would recommend C++17 as a minimum, as it added several new features that makes C++ safer and easier. I would probably just stick to C++20, as it would enable C++ Concepts also.

Reply:
In general, I also prefer more recent versions of the standard. However, some users of HWinfo are using older compiler versions and thus asked for C++11 support. For now, I will stay with this.

The WMIWrapper.h indeed contains some spagetti-code and I am quite sure template specialisation will not resolve this.

I will take a look at the options and benefits of improving this files code when I have some time left...